From 9dad9911bb69b5360ea11fe1802cce5882b332fd Mon Sep 17 00:00:00 2001
From: Jan Sondhauss <jan.sondhauss@wago.com>
Date: Fri, 27 Sep 2019 10:29:43 +0200
Subject: [PATCH] ARM: dts: pfc: refactoring ksz8863 switch

The device tree structure for the ks8863 switch is rearranged in
order to support for "dsa2"-probing. In dsa2 the switch is a child
of the MDIO-bus. In newer kernels dsa2 becomes the default dsa.
By switching to dsa2 we stay compatible current development.
Also dsa2 is used for the marvell switch driver implementation,
which is used in PFC AIO and PFCs with 4 eth ports.

There is now a better destinction between the dsa part and the
swcfg part. Since swcfg is not in mainline and most likely will
never be, this can help getting the ksz8863 driver upstream.

PFCv1 (am35xx) and PFCv3(am33xx) device trees are harmonized for
maintenance reasons.

Signed-off-by: Andreas Schmidt <andreas.schmidt@wago.com>
Signed-off-by: Jan Sondhauss <jan.sondhauss@wago.com>
---
 arch/arm/boot/dts/am335x-pfc-750_8100_8101.dts     |  59 ++++++----
 arch/arm/boot/dts/am335x-pfc-750_8102.dts          |  64 +++++++----
 arch/arm/boot/dts/am335x-pfc-750_810x.dtsi         |  16 ---
 arch/arm/boot/dts/am335x-pfc-750_8208.dts          |  46 +++++---
 arch/arm/boot/dts/am335x-pfc-750_8208.dtsi         |  31 ------
 arch/arm/boot/dts/am335x-pfc-750_8212.dts          |  16 +--
 arch/arm/boot/dts/am335x-pfc-750_8213.dts          |  16 +--
 arch/arm/boot/dts/am335x-pfc-750_8214.dts          |  16 +--
 arch/arm/boot/dts/am335x-pfc-750_8216.dts          |  16 +--
 arch/arm/boot/dts/am335x-pfc-750_8217.dts          | 122 ++++++++++-----------
 arch/arm/boot/dts/am335x-pfc-750_821x-ksz8863.dtsi |  42 -------
 .../boot/dts/am335x-pfc-750_821x-mv88e6321.dtsi    |   2 -
 arch/arm/boot/dts/am335x-pfc-750_821x.dtsi         |   1 -
 arch/arm/boot/dts/am335x-pfc-750_8xxx-ksz8863.dtsi |  68 ++++++++++++
 arch/arm/boot/dts/am335x-pfc.dtsi                  |  10 --
 arch/arm/boot/dts/am335x-pfc100-setupfw.dts        |  19 +++-
 arch/arm/boot/dts/am335x-pfc200v2-setupfw.dts      |  30 +++--
 arch/arm/boot/dts/am335x-pfc200v3-setupfw.dts      |  69 +++++-------
 arch/arm/boot/dts/am3505-pfc-750_8202.dts          |   2 +-
 arch/arm/boot/dts/am3505-pfc-750_8203.dts          |   1 +
 arch/arm/boot/dts/am3505-pfc-750_8204.dts          |   1 +
 arch/arm/boot/dts/am3505-pfc-750_8206.dts          |   1 +
 arch/arm/boot/dts/am3505-pfc-750_8207.dts          |   1 +
 arch/arm/boot/dts/am3505-pfc-750_820x-ksz8863.dtsi |  79 +++++++++++++
 arch/arm/boot/dts/am3505-pfc-750_820x.dtsi         |  54 ---------
 arch/arm/boot/dts/am3505-pfc200-setupfw.dts        |   1 +
 arch/arm/boot/dts/am3xxx-pfc-750_8xxx-ksz8863.dtsi |  68 ------------
 arch/arm/configs/am3xxx_pfc_generic_defconfig      |   1 -
 28 files changed, 410 insertions(+), 442 deletions(-)
 delete mode 100644 arch/arm/boot/dts/am335x-pfc-750_821x-ksz8863.dtsi
 create mode 100644 arch/arm/boot/dts/am335x-pfc-750_8xxx-ksz8863.dtsi
 create mode 100644 arch/arm/boot/dts/am3505-pfc-750_820x-ksz8863.dtsi
 delete mode 100644 arch/arm/boot/dts/am3xxx-pfc-750_8xxx-ksz8863.dtsi

diff --git a/arch/arm/boot/dts/am335x-pfc-750_8100_8101.dts b/arch/arm/boot/dts/am335x-pfc-750_8100_8101.dts
index 893e5c6..45f28a3 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8100_8101.dts
+++ b/arch/arm/boot/dts/am335x-pfc-750_8100_8101.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am335x-pfc-750_810x.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_810x-dip.dtsi"
 #include "wago-devconf.dtsi"
 
@@ -16,31 +17,36 @@
 	compatible = "wago,am335x-pfc-750_810x-0028", "wago,am335x-pfc", "ti,am33xx";
 };
 
-&dsa {
-	status = "okay";
+&am33xx_pinmux {
+	ksz8863_switch_pins: pinmux_ksz8863_switch_pins {
+		pinctrl-single,pins = <
+			AM33XX_MII1_RXD0_GPIO2_21(PIN_OUTPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_GPMC_BEN1_GPIO1_28(PIN_INPUT_PULLUP)
+		>;
+	};
 };
 
-&watchdog {
+&aes {
 	status = "okay";
 };
 
-&oms {
+&bitbang_mdio0 {
 	status = "okay";
 };
 
-&ksz8863 {
+&cppi41dma {
 	status = "okay";
 };
 
-&swcfg_ksz8863 {
+&elm {
 	status = "okay";
 };
 
-&uart4 {
+&ecap0 {
 	status = "okay";
 };
 
-&elm {
+&epwmss0 {
 	status = "okay";
 };
 
@@ -48,46 +54,61 @@
 	status = "okay";
 };
 
-&cppi41dma {
+&i2c0 {
 	status = "okay";
 };
 
-&spi0 {
+&kbus0 {
 	status = "okay";
 };
 
-&kbus0 {
+&ksz8863_switch {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ksz8863_switch_pins>;
+
+	ksz,reset-gpio = <&gpio2 21 GPIO_ACTIVE_LOW>;
+
 	status = "okay";
+
+	ports {
+		port@0 {
+			reg = <1>;
+		};
+
+		port@1 {
+			reg = <2>;
+		};
+	};
 };
 
-&i2c0 {
+&mac {
 	status = "okay";
 };
 
-&mac {
+&mmc1 {
 	status = "okay";
 };
 
-&bitbang_mdio0 {
+&oms {
 	status = "okay";
 };
 
-&epwmss0 {
+&sham {
 	status = "okay";
 };
 
-&ecap0 {
+&spi0 {
 	status = "okay";
 };
 
-&mmc1 {
+&swcfg_ksz8863 {
 	status = "okay";
 };
 
-&sham {
+&uart4 {
 	status = "okay";
 };
 
-&aes {
+&watchdog {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/am335x-pfc-750_8102.dts b/arch/arm/boot/dts/am335x-pfc-750_8102.dts
index 79f741c..a6329ab 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8102.dts
+++ b/arch/arm/boot/dts/am335x-pfc-750_8102.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am335x-pfc-750_810x.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_8xxx-uart1.dtsi"
 #include "am335x-pfc-750_8xxx-uart4.dtsi"
 #include "wago-devconf.dtsi"
@@ -17,82 +18,103 @@
 	compatible = "wago,am335x-pfc-750_810x-000c", "wago,am335x-pfc", "ti,am33xx";
 };
 
-&dsa {
-	status = "okay";
+&am33xx_pinmux {
+	ksz8863_switch_pins: pinmux_ksz8863_switch_pins {
+		pinctrl-single,pins = <
+			AM33XX_MII1_RXD0_GPIO2_21(PIN_OUTPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_GPMC_BEN1_GPIO1_28(PIN_INPUT_PULLUP)
+		>;
+	};
 };
 
-&watchdog {
+&aes {
 	status = "okay";
 };
 
-&oms {
+&bitbang_mdio0 {
 	status = "okay";
 };
 
-&ksz8863 {
+&cppi41dma {
 	status = "okay";
 };
 
-&swcfg_ksz8863 {
+&ecap0 {
 	status = "okay";
 };
 
-&uart4 {
+&elm {
 	status = "okay";
 };
 
-&uart1 {
+&epwmss0 {
 	status = "okay";
 };
 
-&elm {
+&gpmc {
 	status = "okay";
 };
 
-&gpmc {
+&i2c0 {
 	status = "okay";
 };
 
-&cppi41dma {
+&kbus0 {
 	status = "okay";
 };
 
-&spi0 {
+&ksz8863_switch {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ksz8863_switch_pins>;
+
+	ksz,reset-gpio = <&gpio2 21 GPIO_ACTIVE_LOW>;
+
 	status = "okay";
+
+	ports {
+		port@0 {
+			reg = <1>;
+		};
+
+		port@1 {
+			reg = <2>;
+		};
+	};
+
 };
 
-&kbus0 {
+&mac {
 	status = "okay";
 };
 
-&i2c0 {
+&mmc1 {
 	status = "okay";
 };
 
-&mac {
+&oms {
 	status = "okay";
 };
 
-&bitbang_mdio0 {
+&sham {
 	status = "okay";
 };
 
-&epwmss0 {
+&spi0 {
 	status = "okay";
 };
 
-&ecap0 {
+&swcfg_ksz8863 {
 	status = "okay";
 };
 
-&mmc1 {
+&uart1 {
 	status = "okay";
 };
 
-&sham {
+&uart4 {
 	status = "okay";
 };
 
-&aes {
+&watchdog {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/am335x-pfc-750_810x.dtsi b/arch/arm/boot/dts/am335x-pfc-750_810x.dtsi
index 66e02b1..0d72446 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_810x.dtsi
+++ b/arch/arm/boot/dts/am335x-pfc-750_810x.dtsi
@@ -7,7 +7,6 @@
  */
 
 #include "am335x-pfc.dtsi"
-#include "am3xxx-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_8xxx-uart4.dtsi"
 #include "am335x-pfc-750_8xxx-leds.dtsi"
 #include "am335x-pfc-750_8xxx-nvram.dtsi"
@@ -52,14 +51,6 @@
 	gpios = <&gpio0 16 GPIO_ACTIVE_LOW>;
 };
 
-&ksz8863 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&ksz8863_pins>;
-
-	ksz,reset-gpio = <&gpio2 21 GPIO_ACTIVE_LOW>;
-	ksz,disable-internal-ldo;
-};
-
 &gpmc {
 	pinctrl-names = "default";
 	pinctrl-0 = <&gpmc_data_pins>, <&gpmc_addr_pins>, <&gpmc_ctrl_pins>;
@@ -202,13 +193,6 @@
 		>;
 	};
 
-	ksz8863_pins: pinmux_ksz8863_pins {
-		pinctrl-single,pins = <
-			AM33XX_MII1_RXD0_GPIO2_21(PIN_OUTPUT_PULLDOWN | SLEWCTRL_SLOW)
-			AM33XX_GPMC_BEN1_GPIO1_28(PIN_INPUT_PULLUP)
-		>;
-	};
-
 	kbus_pins: pinmux_kbus_pins {
 		pinctrl-single,pins = <
 			AM33XX_GPMC_A7_GPIO1_23(PIN_OUTPUT_PULLUP | SLEWCTRL_SLOW)
diff --git a/arch/arm/boot/dts/am335x-pfc-750_8208.dts b/arch/arm/boot/dts/am335x-pfc-750_8208.dts
index cef1eac..0cb2fbd 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8208.dts
+++ b/arch/arm/boot/dts/am335x-pfc-750_8208.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am335x-pfc-750_8208.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_8xxx-uart1.dtsi"
 #include "am335x-pfc-750_8208-netx.dtsi"
 #include "am335x-pfc-750_82xx-dcan.dtsi"
@@ -18,23 +19,24 @@
 	compatible = "wago,am335x-pfc-750_8208-004e", "wago,am335x-pfc", "ti,am33xx";
 };
 
-&dsa {
-	status = "okay";
-};
-
-&watchdog {
-	status = "okay";
+&am33xx_pinmux {
+	ksz8863_switch_pins: pinmux_ksz8863_switch_pins {
+		pinctrl-single,pins = <
+			AM33XX_MII1_RX_CLK_GPIO3_10(PIN_OUTPUT_PULLDOWN)
+			AM33XX_SPI0_CS1_GPIO0_6(PIN_INPUT_PULLDOWN)
+		>;
+	};
 };
 
-&oms {
+&aes {
 	status = "okay";
 };
 
-&ksz8863 {
+&bitbang_mdio0 {
 	status = "okay";
 };
 
-&swcfg_ksz8863 {
+&cppi41dma {
 	status = "okay";
 };
 
@@ -42,11 +44,11 @@
 	status = "okay";
 };
 
-&cppi41dma {
+&elm {
 	status = "okay";
 };
 
-&spi0 {
+&gpmc {
 	status = "okay";
 };
 
@@ -58,11 +60,17 @@
 	status = "okay";
 };
 
-&mac {
+&ksz8863_switch {
+	pinctrl-names = "default";
+	pintctrl-0 = <&ksz8863_switch_pins>;
+
+	ksz,reset-gpio = <&gpio3 10 GPIO_ACTIVE_LOW>;
+	reg = <0>;
+
 	status = "okay";
 };
 
-&bitbang_mdio0 {
+&mac {
 	status = "okay";
 };
 
@@ -70,19 +78,19 @@
 	status = "okay";
 };
 
-&sham {
+&oms {
 	status = "okay";
 };
 
-&aes {
+&sham {
 	status = "okay";
 };
 
-&elm {
+&spi0 {
 	status = "okay";
 };
 
-&gpmc {
+&swcfg_ksz8863 {
 	status = "okay";
 };
 
@@ -97,3 +105,7 @@
 &uio_netx {
 	status = "okay";
 };
+
+&watchdog {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/am335x-pfc-750_8208.dtsi b/arch/arm/boot/dts/am335x-pfc-750_8208.dtsi
index 7578e6d..7949a57 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8208.dtsi
+++ b/arch/arm/boot/dts/am335x-pfc-750_8208.dtsi
@@ -7,7 +7,6 @@
  */
 
 #include "am335x-pfc.dtsi"
-#include "am3xxx-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_8xxx-uart4.dtsi"
 #include "am335x-pfc-750_8xxx-leds.dtsi"
 #include "am335x-pfc-750_8xxx-nvram.dtsi"
@@ -92,13 +91,6 @@
 		>;
 	};
 
-	ksz8863_pins: pinmux_ksz8863_pins {
-		pinctrl-single,pins = <
-			AM33XX_MII1_RX_CLK_GPIO3_10(PIN_OUTPUT_PULLDOWN)
-			AM33XX_SPI0_CS1_GPIO0_6(PIN_INPUT_PULLDOWN)
-		>;
-	};
-
 	kbus_pins: pinmux_kbus_pins {
 		pinctrl-single,pins = <
 			AM33XX_GPMC_A7_GPIO1_23(PIN_OUTPUT_PULLUP | SLEWCTRL_SLOW)
@@ -190,29 +182,6 @@
 	active_slave = <1>;
 };
 
-&ksz8863 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&ksz8863_pins>;
-
-	ksz,reset-gpio = <&gpio3 10 GPIO_ACTIVE_LOW>;
-	ksz,disable-internal-ldo;
-};
-
-&dsa {
-	switch@0 {
-		/* phys are twisted on hw */
-		port@0 {
-			reg = <2>;
-			phy-handle = <&phy2>;
-		};
-
-		port@1 {
-			reg = <1>;
-			phy-handle = <&phy1>;
-		};
-	};
-};
-
 &gpmc {
 	pinctrl-names = "default";
 	pinctrl-0 = <&gpmc_data_pins>, <&gpmc_addr_pins>, <&gpmc_ctrl_pins>;
diff --git a/arch/arm/boot/dts/am335x-pfc-750_8212.dts b/arch/arm/boot/dts/am335x-pfc-750_8212.dts
index 5da8d40..0ccdd23 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8212.dts
+++ b/arch/arm/boot/dts/am335x-pfc-750_8212.dts
@@ -8,7 +8,7 @@
 /dts-v1/;
 
 #include "am335x-pfc-750_821x.dtsi"
-#include "am335x-pfc-750_821x-ksz8863.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_8xxx-uart1.dtsi"
 #include "wago-devconf.dtsi"
 
@@ -25,10 +25,6 @@
 	status = "okay";
 };
 
-&dsa {
-	status = "okay";
-};
-
 &elm {
 	status = "okay";
 };
@@ -45,11 +41,7 @@
 	status = "okay";
 };
 
-&ksz8863 {
-	status = "okay";
-};
-
-&swcfg_ksz8863 {
+&ksz8863_switch {
 	status = "okay";
 };
 
@@ -77,6 +69,10 @@
 	status = "okay";
 };
 
+&swcfg_ksz8863 {
+	status = "okay";
+};
+
 &uart0 {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/am335x-pfc-750_8213.dts b/arch/arm/boot/dts/am335x-pfc-750_8213.dts
index c96e3de..43892b0 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8213.dts
+++ b/arch/arm/boot/dts/am335x-pfc-750_8213.dts
@@ -8,7 +8,7 @@
 /dts-v1/;
 
 #include "am335x-pfc-750_821x.dtsi"
-#include "am335x-pfc-750_821x-ksz8863.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_82xx-dcan.dtsi"
 #include "wago-devconf.dtsi"
 
@@ -29,10 +29,6 @@
 	status = "okay";
 };
 
-&dsa {
-	status = "okay";
-};
-
 &elm {
 	status = "okay";
 };
@@ -49,11 +45,7 @@
 	status = "okay";
 };
 
-&ksz8863 {
-	status = "okay";
-};
-
-&swcfg_ksz8863 {
+&ksz8863_switch {
 	status = "okay";
 };
 
@@ -81,6 +73,10 @@
 	status = "okay";
 };
 
+&swcfg_ksz8863 {
+	status = "okay";
+};
+
 &uart0 {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/am335x-pfc-750_8214.dts b/arch/arm/boot/dts/am335x-pfc-750_8214.dts
index 32eb8d8..e9b752c 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8214.dts
+++ b/arch/arm/boot/dts/am335x-pfc-750_8214.dts
@@ -9,7 +9,7 @@
 
 #include "am335x-pfc-750_821x.dtsi"
 #include "am335x-pfc-750_8xxx-uart1.dtsi"
-#include "am335x-pfc-750_821x-ksz8863.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_82xx-dcan.dtsi"
 #include "wago-devconf.dtsi"
 
@@ -30,10 +30,6 @@
 	status = "okay";
 };
 
-&dsa {
-	status = "okay";
-};
-
 &elm {
 	status = "okay";
 };
@@ -50,11 +46,7 @@
 	status = "okay";
 };
 
-&ksz8863 {
-	status = "okay";
-};
-
-&swcfg_ksz8863 {
+&ksz8863_switch {
 	status = "okay";
 };
 
@@ -82,6 +74,10 @@
 	status = "okay";
 };
 
+&swcfg_ksz8863 {
+	status = "okay";
+};
+
 &uart0 {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/am335x-pfc-750_8216.dts b/arch/arm/boot/dts/am335x-pfc-750_8216.dts
index 4428724..ed951e3 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8216.dts
+++ b/arch/arm/boot/dts/am335x-pfc-750_8216.dts
@@ -8,7 +8,7 @@
 /dts-v1/;
 
 #include "am335x-pfc-750_821x.dtsi"
-#include "am335x-pfc-750_821x-ksz8863.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_8xxx-uart1.dtsi"
 #include "am335x-pfc-750_82xx-dcan.dtsi"
 #include "am335x-pfc-750_821x-dpc31.dtsi"
@@ -31,10 +31,6 @@
 	status = "okay";
 };
 
-&dsa {
-	status = "okay";
-};
-
 &elm {
 	status = "okay";
 };
@@ -51,11 +47,7 @@
 	status = "okay";
 };
 
-&ksz8863 {
-	status = "okay";
-};
-
-&swcfg_ksz8863 {
+&ksz8863_switch {
 	status = "okay";
 };
 
@@ -83,6 +75,10 @@
 	status = "okay";
 };
 
+&swcfg_ksz8863 {
+	status = "okay";
+};
+
 &uart0 {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/am335x-pfc-750_8217.dts b/arch/arm/boot/dts/am335x-pfc-750_8217.dts
index 5ffc7e5..4575071 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8217.dts
+++ b/arch/arm/boot/dts/am335x-pfc-750_8217.dts
@@ -8,7 +8,7 @@
 /dts-v1/;
 
 #include "am335x-pfc-750_821x.dtsi"
-#include "am335x-pfc-750_821x-ksz8863.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_8xxx-uart1.dtsi"
 #include "wago-devconf.dtsi"
 
@@ -17,61 +17,84 @@
 	compatible = "wago,am335x-pfc-750_821x-1006", "wago,am335x-pfc", "ti,am33xx";
 };
 
-&cppi41dma {
+&aes {
 	status = "okay";
 };
 
-&usb {
+&bitbang_mdio0 {
 	status = "okay";
 };
 
-&usb_ctrl_mod {
+&cppi41dma {
 	status = "okay";
 };
 
-&usb1_phy {
+&elm {
 	status = "okay";
 };
 
-&usb1 {
+&gpmc {
 	status = "okay";
-	dr_mode = "host";
 };
 
-&aes {
+&i2c0 {
 	status = "okay";
 };
 
-&bitbang_mdio0 {
+&kbus0 {
 	status = "okay";
 };
 
-&dsa {
+&ksz8863_switch {
 	status = "okay";
 };
 
-&elm {
-	status = "okay";
+&lb61_0 {
+	label = "s5-red";
 };
 
-&gpmc {
-	status = "okay";
+&lb61_1 {
+	label = "s5-green";
 };
 
-&i2c0 {
-	status = "okay";
+&lb61_2 {
+	label = "s4-red";
 };
 
-&kbus0 {
-	status = "okay";
+&lb61_3 {
+	label = "s4-green";
 };
 
-&ksz8863 {
-	status = "okay";
+&lb61_4 {
+	label = "s3-red";
 };
 
-&swcfg_ksz8863 {
-	status = "okay";
+&lb61_5 {
+	label = "s3-green";
+};
+
+&lb61_6 {
+	label = "s2-red";
+};
+
+&lb61_7 {
+	label = "s2-green";
+};
+
+&lb61_8 {
+	label = "s1-red";
+};
+
+&lb61_9 {
+	label = "s1-green";
+};
+
+&lb61_10 {
+	label = "net-red";
+};
+
+&lb61_11 {
+	label = "net-green";
 };
 
 &mac {
@@ -98,6 +121,10 @@
 	status = "okay";
 };
 
+&swcfg_ksz8863 {
+	status = "okay";
+};
+
 &uart0 {
 	status = "okay";
 };
@@ -110,54 +137,23 @@
 	status = "okay";
 };
 
-&watchdog {
+&usb {
 	status = "okay";
 };
 
-&lb61_0 {
-	label = "s5-red";
-};
-
-&lb61_1 {
-	label = "s5-green";
-};
-
-&lb61_2 {
-	label = "s4-red";
-};
-
-&lb61_3 {
-	label = "s4-green";
-};
-
-&lb61_4 {
-	label = "s3-red";
-};
-
-&lb61_5 {
-	label = "s3-green";
-};
-
-&lb61_6 {
-	label = "s2-red";
-};
-
-&lb61_7 {
-	label = "s2-green";
-};
-
-&lb61_8 {
-	label = "s1-red";
+&usb_ctrl_mod {
+	status = "okay";
 };
 
-&lb61_9 {
-	label = "s1-green";
+&usb1_phy {
+	status = "okay";
 };
 
-&lb61_10 {
-	label = "net-red";
+&usb1 {
+	status = "okay";
+	dr_mode = "host";
 };
 
-&lb61_11 {
-	label = "net-green";
+&watchdog {
+	status = "okay";
 };
diff --git a/arch/arm/boot/dts/am335x-pfc-750_821x-ksz8863.dtsi b/arch/arm/boot/dts/am335x-pfc-750_821x-ksz8863.dtsi
deleted file mode 100644
index 930fca7..0000000
--- a/arch/arm/boot/dts/am335x-pfc-750_821x-ksz8863.dtsi
+++ /dev/null
@@ -1,42 +0,0 @@
-/*
- * Copyright (C) 2019 WAGO Kontakttechnik GmbH & Co. KG - https://www.wago.com/
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-
-#include "am3xxx-pfc-750_8xxx-ksz8863.dtsi"
-
-&am33xx_pinmux {
-	ksz8863_pins: pinmux_ksz8863_pins {
-		pinctrl-single,pins = <
-			AM33XX_LCD_DATA15_GPIO0_11(PIN_INPUT)
-			AM33XX_GPMC_WPN_GPIO0_31(PIN_INPUT)
-			AM33XX_MII1_RX_ER_GPIO3_2(PIN_INPUT)
-		>;
-	};
-};
-
-&ksz8863 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&ksz8863_pins>;
-
-	ksz,reset-gpio = <&io_expander_70 4 GPIO_ACTIVE_LOW>;
-	ksz,disable-internal-ldo;
-};
-
-&dsa {
-	switch@0 {
-		/* phys are twisted on hw */
-		port@0 {
-			reg = <2>;
-			phy-handle = <&phy2>;
-		};
-
-		port@1 {
-			reg = <1>;
-			phy-handle = <&phy1>;
-		};
-	};
-};
diff --git a/arch/arm/boot/dts/am335x-pfc-750_821x-mv88e6321.dtsi b/arch/arm/boot/dts/am335x-pfc-750_821x-mv88e6321.dtsi
index d28baa3..64504ad 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_821x-mv88e6321.dtsi
+++ b/arch/arm/boot/dts/am335x-pfc-750_821x-mv88e6321.dtsi
@@ -100,14 +100,12 @@
 			port@3 {
 				reg = <3>;
 				label = "ethX11";
-				phy-handle = <&int_phy3>;
 				phy-mode = "rmii";
 			};
 
 			port@4 {
 				reg = <4>;
 				label = "ethX12";
-				phy-handle = <&int_phy4>;
 				phy-mode = "rmii";
 			};
 
diff --git a/arch/arm/boot/dts/am335x-pfc-750_821x.dtsi b/arch/arm/boot/dts/am335x-pfc-750_821x.dtsi
index e9a796e..3e46359 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_821x.dtsi
+++ b/arch/arm/boot/dts/am335x-pfc-750_821x.dtsi
@@ -378,4 +378,3 @@
 		line-name = "boot_select";
 	};
 };
-
diff --git a/arch/arm/boot/dts/am335x-pfc-750_8xxx-ksz8863.dtsi b/arch/arm/boot/dts/am335x-pfc-750_8xxx-ksz8863.dtsi
new file mode 100644
index 0000000..ad7cc80
--- /dev/null
+++ b/arch/arm/boot/dts/am335x-pfc-750_8xxx-ksz8863.dtsi
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2019 WAGO Kontakttechnik GmbH & Co. KG - https://www.wago.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+
+/ {
+	swcfg_ksz8863: swcfg_ksz8863 {
+		compatible = "swcfg,ksz8863";
+		swcfg,mii-bus = <&bitbang_mdio0>;
+		swcfg,alias = "ksz8863";
+		swcfg,cpu_port = <2>;
+		swcfg,ports = <3>;
+		swcfg,vlans = <16>;
+		swcfg,switch = <&ksz8863_switch>;
+
+		status = "disabled";
+	};
+};
+
+&bitbang_mdio0 {
+	ksz8863_switch: switch@0 {
+		compatible = "micrel,ksz8863";
+
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ksz,reset-gpio = <&io_expander_70 4 GPIO_ACTIVE_LOW>;
+
+		reg = <0>;
+		dsa,member = <0 0>;
+
+		ksz,reset-switch;
+		ksz,disable-internal-ldo;
+
+		status = "disabled";
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <2>;
+				label = "ethX1";
+				phy-pwrdown;
+			};
+
+			port@1 {
+				reg = <1>;
+				label = "ethX2";
+				phy-pwrdown;
+			};
+
+			port@2 {
+				reg = <3>;
+				label = "cpu";
+				ethernet = <&mac>;
+				fixed-link {
+					speed = <100>;
+					full-duplex;
+				};
+			};
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/am335x-pfc.dtsi b/arch/arm/boot/dts/am335x-pfc.dtsi
index 7552555..605e45c 100644
--- a/arch/arm/boot/dts/am335x-pfc.dtsi
+++ b/arch/arm/boot/dts/am335x-pfc.dtsi
@@ -290,16 +290,6 @@
 	pinctrl-1 = <&gpio_bitbang_mdio_sleep_pins>;
 	gpios = <&gpio0 1 0   /* 0: mdc  */
 		 &gpio0 0 0>; /* 1: mdio */
-
-	phy1: ethernet-phy@1 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <1>;
-	};
-
-	phy2: ethernet-phy@2 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <2>;
-	};
 };
 
 /*
diff --git a/arch/arm/boot/dts/am335x-pfc100-setupfw.dts b/arch/arm/boot/dts/am335x-pfc100-setupfw.dts
index f0435fa..0b2aafa 100644
--- a/arch/arm/boot/dts/am335x-pfc100-setupfw.dts
+++ b/arch/arm/boot/dts/am335x-pfc100-setupfw.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am335x-pfc-750_810x.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_8xxx-uart1.dtsi"
 #include "am335x-pfc-750_810x-musb.dtsi"
 #include "am335x-pfc-750_810x-cmtests.dtsi"
@@ -58,6 +59,13 @@
 			0x1E8 (PIN_OUTPUT | MUX_MODE7)  /* emu1.gpio3_8 */
 		>;
 	};
+
+	ksz8863_switch_pins: pinmux_ksz8863_switch_pins {
+		pinctrl-single,pins = <
+			AM33XX_MII1_RXD0_GPIO2_21(PIN_OUTPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_GPMC_BEN1_GPIO1_28(PIN_INPUT_PULLUP)
+		>;
+	};
 };
 
 &aes {
@@ -72,10 +80,6 @@
 	status = "okay";
 };
 
-&dsa {
-	status = "okay";
-};
-
 &ecap0 {
 	status = "okay";
 };
@@ -100,7 +104,12 @@
 	status = "okay";
 };
 
-&ksz8863 {
+&ksz8863_switch {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ksz8863_switch_pins>;
+
+	ksz,reset-gpio = <&gpio2 21 GPIO_ACTIVE_LOW>;
+
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/am335x-pfc200v2-setupfw.dts b/arch/arm/boot/dts/am335x-pfc200v2-setupfw.dts
index 4975e05..f3e2c38 100644
--- a/arch/arm/boot/dts/am335x-pfc200v2-setupfw.dts
+++ b/arch/arm/boot/dts/am335x-pfc200v2-setupfw.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am335x-pfc-750_8208.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "am335x-pfc-750_8208-netx.dtsi"
 #include "am335x-pfc-750_82xx-dcan.dtsi"
 #include "wago-devconf.dtsi"
@@ -35,6 +36,16 @@
 	};
 };
 
+&am33xx_pinmux {
+
+	ksz8863_switch_pins: pinmux_ksz8863_switch_pins {
+		pinctrl-single,pins = <
+			AM33XX_MII1_RX_CLK_GPIO3_10(PIN_OUTPUT_PULLDOWN)
+			AM33XX_SPI0_CS1_GPIO0_6(PIN_INPUT_PULLDOWN)
+		>;
+	};
+};
+
 &aes {
 	status = "okay";
 };
@@ -51,10 +62,6 @@
 	status = "okay";
 };
 
-&dsa {
-	status = "okay";
-};
-
 &elm {
 	status = "okay";
 };
@@ -71,7 +78,12 @@
 	status = "okay";
 };
 
-&ksz8863 {
+&ksz8863_switch {
+	pinctrl-names = "default";
+	pintctrl-0 = <&ksz8863_switch_pins>;
+
+	ksz,reset-gpio = <&gpio3 10 GPIO_ACTIVE_LOW>;
+
 	status = "okay";
 };
 
@@ -83,16 +95,16 @@
 	status = "okay";
 };
 
-&swcfg_ksz8863 {
-	swcfg,alias = "ksz8863";
+&sham {
 	status = "okay";
 };
 
-&sham {
+&spi0 {
 	status = "okay";
 };
 
-&spi0 {
+&swcfg_ksz8863 {
+	swcfg,alias = "ksz8863";
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/am335x-pfc200v3-setupfw.dts b/arch/arm/boot/dts/am335x-pfc200v3-setupfw.dts
index 1352a9a..90ffbe2 100644
--- a/arch/arm/boot/dts/am335x-pfc200v3-setupfw.dts
+++ b/arch/arm/boot/dts/am335x-pfc200v3-setupfw.dts
@@ -8,7 +8,7 @@
 /dts-v1/;
 
 #include "am335x-pfc-750_821x.dtsi"
-#include "am335x-pfc-750_821x-ksz8863.dtsi"
+#include "am335x-pfc-750_8xxx-ksz8863.dtsi"
 #include "wago-devconf.dtsi"
 
 / {
@@ -140,16 +140,6 @@
 				};
 			};
 
-			/*port@2 {
-				reg = <2>;
-				label = "cpu";
-				ethernet = <&mac>;
-				fixed-link {
-					speed = <100>;
-					full-duplex;
-				};
-			};*/
-
 			port@3 {
 				reg = <3>;
 				label = "ethX11";
@@ -179,7 +169,7 @@
 	};
 };
 
-&dsa {
+&cppi41dma {
 	status = "okay";
 };
 
@@ -202,38 +192,13 @@
 /* we assume that the bootloader has already resetted the
  * switch
  */
-&ksz8863 {
+&ksz8863_switch {
 	/delete-property/ ksz,reset-switch;
 	/delete-property/ ksz,reset-gpio;
 
 	status = "okay";
 };
 
-&mac {
-	status = "okay";
-};
-
-&mmc1 {
-	status = "okay";
-};
-
-&mmc2 {
-	status = "okay";
-};
-
-&sham {
-	status = "okay";
-};
-
-&spi0 {
-	status = "okay";
-};
-
-&swcfg_ksz8863 {
-	swcfg,alias = "ksz8863";
-	status = "okay";
-};
-
 &lb61_0 {
 	label = "bf-red";
 };
@@ -258,11 +223,32 @@
 	label = "can-green";
 };
 
-&uart5 {
+&mac {
 	status = "okay";
 };
 
-&watchdog {
+&mmc1 {
+	status = "okay";
+};
+
+&mmc2 {
+	status = "okay";
+};
+
+&sham {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+};
+
+&swcfg_ksz8863 {
+	swcfg,alias = "ksz8863";
+	status = "okay";
+};
+
+&uart5 {
 	status = "okay";
 };
 
@@ -283,7 +269,6 @@
 	dr_mode = "host";
 };
 
-&cppi41dma {
+&watchdog {
 	status = "okay";
 };
-
diff --git a/arch/arm/boot/dts/am3505-pfc-750_8202.dts b/arch/arm/boot/dts/am3505-pfc-750_8202.dts
index 79c1f06..c18b331 100644
--- a/arch/arm/boot/dts/am3505-pfc-750_8202.dts
+++ b/arch/arm/boot/dts/am3505-pfc-750_8202.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am3505-pfc-750_820x.dtsi"
+#include "am3505-pfc-750_820x-ksz8863.dtsi"
 #include "am3505-pfc-750_820x-uart1.dtsi"
 #include "wago-devconf.dtsi"
 
@@ -16,4 +17,3 @@
 	compatible = "wago,am3505-pfc-750_820x-000c", "wago,am3505-pfc", "ti,am3517-evm", "ti,am3517", "ti,omap3";
 
 };
-
diff --git a/arch/arm/boot/dts/am3505-pfc-750_8203.dts b/arch/arm/boot/dts/am3505-pfc-750_8203.dts
index 5856cab..034cf64 100644
--- a/arch/arm/boot/dts/am3505-pfc-750_8203.dts
+++ b/arch/arm/boot/dts/am3505-pfc-750_8203.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am3505-pfc-750_820x.dtsi"
+#include "am3505-pfc-750_820x-ksz8863.dtsi"
 #include "am3505-pfc-750_820x-can.dtsi"
 #include "wago-devconf.dtsi"
 
diff --git a/arch/arm/boot/dts/am3505-pfc-750_8204.dts b/arch/arm/boot/dts/am3505-pfc-750_8204.dts
index 9c03962..23421a2 100644
--- a/arch/arm/boot/dts/am3505-pfc-750_8204.dts
+++ b/arch/arm/boot/dts/am3505-pfc-750_8204.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am3505-pfc-750_820x.dtsi"
+#include "am3505-pfc-750_820x-ksz8863.dtsi"
 #include "am3505-pfc-750_820x-uart1.dtsi"
 #include "am3505-pfc-750_820x-can.dtsi"
 #include "wago-devconf.dtsi"
diff --git a/arch/arm/boot/dts/am3505-pfc-750_8206.dts b/arch/arm/boot/dts/am3505-pfc-750_8206.dts
index 319be3f..b5f3432 100644
--- a/arch/arm/boot/dts/am3505-pfc-750_8206.dts
+++ b/arch/arm/boot/dts/am3505-pfc-750_8206.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am3505-pfc-750_820x.dtsi"
+#include "am3505-pfc-750_820x-ksz8863.dtsi"
 #include "am3505-pfc-750_820x-uart1.dtsi"
 #include "am3505-pfc-750_820x-can.dtsi"
 #include "am3505-pfc-750_820x-dpc31.dtsi"
diff --git a/arch/arm/boot/dts/am3505-pfc-750_8207.dts b/arch/arm/boot/dts/am3505-pfc-750_8207.dts
index 11c39e6..4187005 100644
--- a/arch/arm/boot/dts/am3505-pfc-750_8207.dts
+++ b/arch/arm/boot/dts/am3505-pfc-750_8207.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am3505-pfc-750_820x.dtsi"
+#include "am3505-pfc-750_820x-ksz8863.dtsi"
 #include "am3505-pfc-750_820x-uart1.dtsi"
 #include "am3505-pfc-750_820x-musb.dtsi"
 #include "wago-devconf.dtsi"
diff --git a/arch/arm/boot/dts/am3505-pfc-750_820x-ksz8863.dtsi b/arch/arm/boot/dts/am3505-pfc-750_820x-ksz8863.dtsi
new file mode 100644
index 0000000..df202c1
--- /dev/null
+++ b/arch/arm/boot/dts/am3505-pfc-750_820x-ksz8863.dtsi
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2019 WAGO Kontakttechnik GmbH & Co. KG - https://www.wago.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/ {
+	swcfg_ksz8863: swcfg_ksz8863 {
+		compatible = "swcfg,ksz8863";
+		swcfg,mii-bus = <&bitbang_mdio0>;
+		swcfg,alias = "ksz8863";
+		swcfg,cpu_port = <2>;
+		swcfg,ports = <3>;
+		swcfg,vlans = <16>;
+		swcfg,switch = <&ksz8863_switch>;
+
+		status = "okay";
+	};
+};
+
+&omap3_pmx_core {
+	ksz8863_pins: pinmux_ksz8863_pins {
+		pinctrl-single,pins = <
+			0x134 (PIN_OUTPUT | MUX_MODE4) /* mmc2_dat4.gpio5_8 (gpio_136) nrst_switch */
+			OMAP3_CORE1_IOPAD(0x2168, PIN_INPUT | MUX_MODE4) /* mmc2_dat6.gpio5_10 (gpio_138) nintswitch */
+		>;
+	};
+};
+
+&bitbang_mdio0 {
+	ksz8863_switch: switch@0 {
+		compatible = "micrel,ksz8863";
+
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&ksz8863_pins>;
+
+		ksz,reset-gpio = <&gpio5 8 GPIO_ACTIVE_LOW>;	/* gpio2_21: RMII2.EN-PHY */
+
+		reg = <0>;
+		dsa,member = <0 0>;
+		dsa,enable-on-boot;
+
+		ksz,reset-switch;
+
+		status = "okay";
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <2>;
+				label = "ethX1";
+				phy-pwrdown;
+			};
+
+			port@1 {
+				reg = <1>;
+				label = "ethX2";
+				phy-pwrdown;
+			};
+
+			port@2 {
+				reg = <3>;
+				label = "cpu";
+				ethernet = <&davinci_emac>;
+				fixed-link {
+					speed = <100>;
+					full-duplex;
+				};
+			};
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/am3505-pfc-750_820x.dtsi b/arch/arm/boot/dts/am3505-pfc-750_820x.dtsi
index afaea28..7aa18c5 100644
--- a/arch/arm/boot/dts/am3505-pfc-750_820x.dtsi
+++ b/arch/arm/boot/dts/am3505-pfc-750_820x.dtsi
@@ -7,7 +7,6 @@
  */
 
 #include "am3517.dtsi"
-#include "am3xxx-pfc-750_8xxx-ksz8863.dtsi"
 
 / {
 	compatible = "wago,am3505-pfc", "ti,am3517", "ti,omap3";
@@ -279,42 +278,6 @@
 	};
 };
 
-&ksz8863 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&ksz8863_pins>;
-
-	ksz,reset-gpio = <&gpio5 8 GPIO_ACTIVE_LOW>;	/* gpio2_21: RMII2.EN-PHY */
-
-	status = "okay";
-};
-
-&swcfg_ksz8863 {
-	status = "okay";
-};
-
-&dsa {
-	status = "okay";
-	dsa,ethernet = <&davinci_emac>; /* TODO: should we use 'cpsw_emac1' instead? */
-
-	switch@0 {
-		port@0 {
-			reg = <2>;
-			phy-handle = <&phy2>;
-		};
-
-		port@1 {
-			reg = <1>;
-			phy-handle = <&phy1>;
-		};
-
-		port@2 {
-			reg = <3>;
-			label = "cpu";
-		};
-	};
-};
-
-
 &omap3_pmx_core {
 	pinctrl-names = "default";
 	pinctrl-0 = <
@@ -388,13 +351,6 @@
 		>;
 	};
 
-	ksz8863_pins: pinmux_ksz8863_pins {
-		pinctrl-single,pins = <
-			0x134 (PIN_OUTPUT | MUX_MODE4) /* mmc2_dat4.gpio5_8 (gpio_136) nrst_switch */
-			OMAP3_CORE1_IOPAD(0x2168, PIN_INPUT | MUX_MODE4) /* mmc2_dat6.gpio5_10 (gpio_138) nintswitch */
-		>;
-	};
-
 	emac_pins: pinmux_emac_pins {
 		pinctrl-single,pins = <
 			0x1d2 (PIN_INPUT | MUX_MODE0)           /* rmii_rxd0.rmii_rxd0           */
@@ -502,16 +458,6 @@
 	pinctrl-1 = <&gpio_bitbang_mdio_sleep_pins>;
 	gpios = <&gpio4 12 0	/* 0: mdc  */
 		&gpio4 11 0>;   /* 1: mdio */
-
-	phy1: ethernet-phy@1 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <1>;
-	};
-
-	phy2: ethernet-phy@2 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <2>;
-	};
 };
 
 &davinci_emac {
diff --git a/arch/arm/boot/dts/am3505-pfc200-setupfw.dts b/arch/arm/boot/dts/am3505-pfc200-setupfw.dts
index f79a33b..1d06fe9 100644
--- a/arch/arm/boot/dts/am3505-pfc200-setupfw.dts
+++ b/arch/arm/boot/dts/am3505-pfc200-setupfw.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "am3505-pfc-750_820x.dtsi"
+#include "am3505-pfc-750_820x-ksz8863.dtsi"
 #include "am3505-pfc-750_820x-uart1.dtsi"
 #include "am3505-pfc-750_820x-can.dtsi"
 #include "am3505-pfc-750_820x-dpc31.dtsi"
diff --git a/arch/arm/boot/dts/am3xxx-pfc-750_8xxx-ksz8863.dtsi b/arch/arm/boot/dts/am3xxx-pfc-750_8xxx-ksz8863.dtsi
deleted file mode 100644
index 9391f5a..0000000
--- a/arch/arm/boot/dts/am3xxx-pfc-750_8xxx-ksz8863.dtsi
+++ /dev/null
@@ -1,68 +0,0 @@
-/*
- * Copyright (C) 2019 WAGO Kontakttechnik GmbH & Co. KG - https://www.wago.com/
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- */
-
-/ {
-	ksz8863: ksz8863 {
-		compatible = "micrel,ksz8863";
-		ksz,reset-switch;
-		ksz,reset-gpio = <&gpio2 21 GPIO_ACTIVE_LOW>;	/* gpio2_21: RMII2.EN-PHY */
-		//ksz,disable-internal-ldo;
-		ksz,mii-bus = <&bitbang_mdio0>;
-
-		status = "disabled";
-	};
-
-	swcfg_ksz8863: swcfg_ksz8863 {
-		compatible = "swcfg,ksz8863";
-		swcfg,mii-bus = <&bitbang_mdio0>;
-		swcfg,alias = "switch0";
-		swcfg,cpu_port = <2>;
-		swcfg,ports = <3>;
-		swcfg,vlans = <16>;
-
-		status = "disabled";
-	};
-
-	dsa: dsa@0 {
-		compatible = "marvell,dsa";
-		#address-cells = <2>;
-		#size-cells = <0>;
-
-		/* interrupts = <10>; --> not supported from the driver! */
-
-		/* master netdevice (eth0) */
-		dsa,ethernet = <&mac>;
-		dsa,mii-bus = <&bitbang_mdio0>;
-		status = "disabled";
-
-		switch@0 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <16 0>;	/* MDIO address 16, switch 0 in tree */
-
-			port@0 {
-				reg = <1>;
-				label = "ethX1";
-				phy-handle = <&phy1>;
-				phy-pwrdown;
-			};
-
-			port@1 {
-				reg = <2>;
-				label = "ethX2";
-				phy-handle = <&phy2>;
-				phy-pwrdown;
-			};
-
-			port@2 {
-				reg = <3>;
-				label = "cpu";
-			};
-		};
-	};
-};
diff --git a/arch/arm/configs/am3xxx_pfc_generic_defconfig b/arch/arm/configs/am3xxx_pfc_generic_defconfig
index defa849..981526d 100644
--- a/arch/arm/configs/am3xxx_pfc_generic_defconfig
+++ b/arch/arm/configs/am3xxx_pfc_generic_defconfig
@@ -308,7 +308,6 @@ CONFIG_TI_CPTS=y
 CONFIG_SWCONFIG=y
 CONFIG_MDIO_BITBANG=y
 CONFIG_MDIO_GPIO=y
-CONFIG_MICREL_KSZ8863_PHY=y
 CONFIG_INTEL_XWAY_PHY=y
 CONFIG_MARVELL_PHY=y
 CONFIG_MICREL_PHY=y
-- 
2.7.4

